backend/src/layer/data/database/mealplan_management.rs

```diff
@@ -862,23 +955,68 @@ mod test {
         assert_eq!(selection.position as u32, pos);
     }
 
-    #[sqlx::test(fixtures("meal", "allergen", "additive"))]
+    #[sqlx::test(fixtures("meal", "allergen", "additive", "nutrition_data", "environment_info"))]
     async fn test_update_food(pool: PgPool) {
         let req = PersistentMealplanManagementData { pool: pool.clone() };
 
         let food_id = Uuid::parse_str("f7337122-b018-48ad-b420-6202dc3cb4ff").unwrap();
         let name = "TEST_FOOD";
+        let nutrition_data = NutritionData {
+            energy: 1,
+            protein: 2,
+            carbohydrates: 3,
+            sugar: 4,
+            fat: 5,
+            saturated_fat: 6,
+            salt: 7,
+        };
+        let environment_info = ParseEnvironmentInfo {
+            co2_rating: 1,
+            co2_value: 2,
+            water_rating: 3,
+            water_value: 4,
+            animal_welfare_rating: 5,
+            rainforest_rating: 6,
+            max_rating: 7,
+        };
 
-        let res = req.update_food(food_id, name).await;
+        let res = req
+            .update_food(
+                food_id,
+                name,
+                Some(nutrition_data.clone()),
+                Some(environment_info.clone()),
+            )
+            .await;
         assert!(res.is_ok());
 
         let selections = sqlx::query!(r#"SELECT name FROM food WHERE food_id = $1"#, food_id)
             .fetch_all(&pool)
             .await
             .unwrap();
-        let selection = selections.first().unwrap();
+        let db_res_name = &selections.first().unwrap().name;
+        let selections = sqlx::query!(
+            r#"SELECT * FROM food_nutrition_data WHERE food_id = $1"#,
+            food_id
+        )
+        .fetch_all(&pool)
+        .await
+        .unwrap();
+        let db_res_salt = selections.first().unwrap().salt;
+        let selections = sqlx::query!(
+            r#"SELECT * FROM food_env_score WHERE food_id = $1"#,
+            food_id
+        )
+        .fetch_all(&pool)
+        .await
+        .unwrap();
+        let db_res_rating = selections.first().unwrap().rainforest_rating;
 
-        assert_eq!(selection.name, name);
+        assert_eq!(db_res_name, name);
+        assert_eq!(db_res_rating, environment_info.rainforest_rating as i32);
+        assert_eq!(db_res_salt, nutrition_data.salt as i32);
+        // tbd: checking env and nutrition data
+        // This could be done better as it is now.
```

Comment:
There is a way to get complete structs without manually parsing out of the query (query-scalar?). This is something that I want to change for this and all other related tests.

Reply:
Do you mean `query_as`?
https://docs.rs/sqlx/latest/sqlx/macro.query_as.html